frontend/model/logger.js

```diff
@@ -0,0 +1,100 @@
+import sbp from '@sbp/sbp'
+import { CAPTURED_LOGS } from '~/frontend/utils/events.js'
+import CircularList from '~/shared/CircularList.js'
+
+const loggingLevels = ['debug', 'error', 'info', 'log', 'warn']
+const originalConsole = console
+const noop = (...args: any) => undefined
+
+export async function createLogger (config: Object, { getItem, removeItem, setItem }: { getItem: Function, removeItem: Function, setItem: Function }): Object {
+  const entries = new CircularList(config.maxEntries)
+  const methods = Object.fromEntries(loggingLevels.map((name) =>
+    [name, (...args) => {
+      originalConsole[name](...args)
+      captureLogEntry(logger, name, config.source, ...args)
+    }]
+  ))
+  const appLogsFilter: string[] = []
+  const consoleProxy = new Proxy({ ...methods }, {
+    get (o, p, r) {
+      return Reflect.has(o, p) ? Reflect.get(o, p, r) : Reflect.get(originalConsole, p, r)
+    },
+    has (o, p) {
+      return Reflect.has(originalConsole, p)
+    }
+  })
+  const logger = {
+    get appLogsFilter () {
+      return [...appLogsFilter]
+    },
+    console: consoleProxy,
+    entries,
+    async clear () {
+      await removeItem('entries')
+      entries.clear()
+    },
+    async save () {
+      try {
+        await setItem('entries', this.entries.toArray())
+      } catch (error) {
+        consoleProxy.error(error)
+      }
+    },
+    setAppLogsFilter (filter: Array<string>) {
+      appLogsFilter.splice(0, appLogsFilter.length, ...filter)
+      // NOTE: Find a way to capture logs without messing up with log file location.
+      // console.log() doesnt include stack trace, so when logged, we can't access
+      // where the log came from (file name), which} difficults debugging if needed.
+      for (const level of loggingLevels) {
+        // $FlowFixMe
+        consoleProxy[level] = appLogsFilter.includes(level) ? methods[level] : noop
+      }
+    }
+  }
+
+  const previousEntries = JSON.parse(await getItem('entries') ?? '[]')
```

Comment:
Claude isn't happy about unhandled error from `JSON.parse`, suggests:

```js
// In the initialization code:
const previousEntries = (() => {
  try {
    const stored = await getItem('entries')
    return stored ? JSON.parse(stored) : []
  } catch (error) {
    consoleProxy.error('Failed to parse stored entries:', error)
    return []
  }
})()
```

Reply:
At a quick glance (right off to bed), Claude's version is more correct but more verbose. Whether it's worth it or not depends on the probability of `getItem` throwing or returning invalid JSON.